perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -4192,34 +4213,33 @@ private void setIfNotEmpty(Map<String, String> map, String value, String key) {
 	 * @param form application form
 	 * @return RegistrarModule if present or null
 	 */
-	private RegistrarModule getRegistrarModule(ApplicationForm form) {
-
+	private List<RegistrarModule> getRegistrarModules(ApplicationForm form) {
```

Comment:
Set here as well, otherwise we would for example try to reject the same application twice, no?